go-controller/pkg/node/gateway_iptables.go

```diff
@@ -328,16 +329,24 @@ func initLocalGatewayNATRules(ifname string, cidr *net.IPNet) error {
 	return addIptRules(getLocalGatewayNATRules(ifname, cidr))
 }
 
+func addChaintoTable(ipt util.IPTablesHelper, tableName, chain string) {
+	if err := ipt.NewChain(tableName, chain); err != nil {
+		klog.V(5).Infof("Chain: \"%s\" in table: \"%s\" already exists, skipping creation: %v", chain, tableName, err)
+	}
+}
+
 func handleGatewayIPTables(iptCallback func(rules []iptRule) error, genGatewayChainRules func(chain string, proto iptables.Protocol) []iptRule) error {
 	rules := make([]iptRule, 0)
-	for _, chain := range []string{iptableNodePortChain, iptableExternalIPChain} {
+	// (NOTE: Order is important, add jump to iptableETPChain before jump to NP/EIP chains)
+	for _, chain := range []string{iptableITPChain, iptableNodePortChain, iptableExternalIPChain, iptableETPChain} {
```

Comment:
can u help me understand the order , since they are separate chains then there should be ordering dependency ?

Reply:
yeah there are two pairs of dependencies:
1)  `OVN-KUBE-ETP` and `OVN-KUBE-NODEPORT`
2) `OVN-KUBE-ETP` and `OVN-KUBE-EXTERNALIP`
This is because both are called from `PRE-ROUTING` chain. So when ETP=Local, we need the `ETP` chain to get priority over the other two chains (NP/EIP) and hence we add the jump to ETP _before_ jump to NP/EIP from PREROUTING.